### PR TITLE
Feat/login alumno profesor responsable

### PR DESCRIPTION
**Pull request by al-martin**
Login implemented and integrated with frontend. Other pages still not working

### CODE REVIEW

File: educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/aux/Security.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	}
+	
+	public static String getHash(String password, String salt) {
+		try {
+			MessageDigest md = MessageDigest.getInstance("SHA-512");
+			md.reset();
+			byte[] hashedPassword = md.digest((salt + password).getBytes(StandardCharsets.UTF_8));
+			System.out.println("Setted password: '" + password + "'");
+			System.out.println("Setted salt: '" + salt + "'");
+			System.out.println("Setted hash: '" + hashedPassword + "'");
```
Comment (d3vv3) [inline educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/aux/Security.java:37]: remember to remove prints on release

File: educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/aux/Security.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	public static Boolean checkPassword(String passwordToCheck, String hash, String salt) {
+		try {
+			MessageDigest md = MessageDigest.getInstance("SHA-512");
+			md.reset();
+			System.out.println("Salt: '" + salt + "'");
+			byte[] hashedPasswordToCheck = md.digest((salt + passwordToCheck).getBytes(StandardCharsets.UTF_8));
+			String hexHashToCheck = Security.bytesToHex(hashedPasswordToCheck);
+			System.out.println("Introduced password: '" + passwordToCheck + "'");
+			System.out.println("Computed hash: '" + hexHashToCheck + "'");
+			System.out.println("Expected hash: '" + hash.toString() + "'");
```
Comment (d3vv3) [inline educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/aux/Security.java:56]: same here

File: educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/model/Usuario.java [hunk 3]
```diff
@@ -19,83 +21,65 @@
 	@Column(nullable = false)
 	private String nombre;
 	@Column(nullable = false)
-	private byte[] hash;
+	private String hash;
 	@Column(nullable = false)
-	private byte[] salt;
+	private String salt;
```
Comment (d3vv3) [inline educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/model/Usuario.java:26]: well played

File: educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/model/Usuario.java [hunk 4]
```diff
@@ -19,83 +21,65 @@
...
-	public Usuario(Integer id, String nombre, String password) {
+	public Usuario(String nombre, String hash, String salt) {
 		super();
 		this.nombre = nombre;
-		this.setPassword(password);
+		this.hash = hash;
+		this.salt = salt;
+		System.out.println("User name: '" + this.nombre + "'");
+		System.out.println("User salt: '" + this.salt.toString() + "'");
+		System.out.println("User hash: '" + this.hash.toString() + "'");
```
Comment (d3vv3) [inline educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/model/Usuario.java:40]: same here

File: educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/rest/CorsFilter.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+@Provider
+public class CorsFilter implements ContainerResponseFilter {
+
+	@Override
+	public void filter(ContainerRequestContext requestContext, ContainerResponseContext responseContext)
+			throws IOException {
+		responseContext.getHeaders().add("Access-Control-Allow-Origin", "*");
+		responseContext.getHeaders().add("Access-Control-Allow-Credentials", "true");
+		responseContext.getHeaders().add("Access-Control-Allow-Headers", "origin, content-type, accept, authorization");
+		responseContext.getHeaders().add("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, OPTIONS, HEAD");
```
Comment (d3vv3) [inline educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/rest/CorsFilter.java:19]: thx cors

**Review by d3vv3 — APPROVED** [review]
good job

**Review by Jaavier8 — APPROVED** [review]
Lgtm

File: educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/model/Usuario.java (hunk 3 above)
Comment (Jaavier8) [reply educovid-backend/educovid-backend/src/main/java/es/upm/dit/isst/educovid/model/Usuario.java:26]: 😂

**Pull request merged**